lib/Transforms/Scalar/CMakeLists.txt

```diff
@@ -54,6 +54,8 @@ add_llvm_library(LLVMScalarOpts
   StraightLineStrengthReduce.cpp
   StructurizeCFG.cpp
   TailRecursionElimination.cpp
+  ../../SYCL/ItaniumDemangle.cpp
+  ../../SYCL/SYCLKernel.cpp
```

Comment:
It is not clear to me why you need this in Transforms/Scalar

Reply:
Ah, I realize that you made some change to some `Transforms/Scalar` passes...
Could it be possible to have only changes in `lib/SYCL` ?
What is the trade-off ?

Reply:
From your idea, try to execute loop idiom pass after SYCL argument flattening pass for the host. Thanks!